packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx

```diff
@@ -43,20 +296,52 @@ const baseRenderLeaf = (props: RenderLeafProps, modifiers: ModifiersStore) => {
   return <span {...props.attributes}>{wrappedChildren}</span>;
 };
 
-const baseRenderElement = (props: RenderElementProps, blocks: BlocksStore) => {
+type BaseRenderElementProps = Direction & {
+  props: RenderElementProps['children'];
+  blocks: BlocksStore;
+  editor: Editor;
+};
+
+const isLink = (element: Element): element is Block<'link'> => {
+  return element.type === 'link';
+};
+
+const isList = (element: Element): element is Block<'list'> => {
+  return element.type === 'list';
+};
+
+const baseRenderElement = ({
+  props,
+  blocks,
+  editor,
+  setDirection,
+  dragDirection,
+}: BaseRenderElementProps) => {
   const blockMatch = Object.values(blocks).find((block) => block.matchNode(props.element));
   const block = blockMatch || blocks.paragraph;
+  const nodePath = ReactEditor.findPath(editor, props.element);
 
-  return block.renderElement(props);
+  // Link is inline block so it cannot be dragged
+  // List is skipped from dragged items
+  if (isLink(props.element) || isList(props.element)) return block.renderElement(props);
+
+  return (
+    <DragAndDropElement index={nodePath} setDirection={setDirection} dragDirection={dragDirection}>
+      {block.renderElement(props)}
+    </DragAndDropElement>
+  );
 };
 
 interface BlocksInputProps {
   placeholder?: string;
 }
 
 const BlocksContent = ({ placeholder }: BlocksInputProps) => {
-  const { editor, disabled, blocks, modifiers } = useBlocksEditorContext('BlocksContent');
+  const { editor, disabled, blocks, modifiers, setLiveText } =
+    useBlocksEditorContext('BlocksContent');
   const blocksRef = React.useRef<HTMLDivElement>(null);
+  const { formatMessage } = useIntl();
+  const [dragDirection, setDirection] = React.useState<DragDirection>('upwards');
```

Comment:
How come this is all the way here? you then pass it down 2/3 levels which feels odd? also why is the default direction upwards? the default is "we're not moving" no?